pkg/minikube/cluster/cluster_provisioner.go

```diff
@@ -0,0 +1,293 @@
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/docker/machine/libmachine/provision/pkgaction"
+	"github.com/docker/machine/libmachine/swarm"
+	"github.com/docker/machine/libmachine/provision"
+
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	engineConfigTemplateRHEL = `[Unit]
+Description=Docker Application Container Engine
+After=network.target rc-local.service
+Requires=rhel-push-plugin.socket
+
+[Service]
+Type=notify
+ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:{{.DockerPort}} -H unix:///var/run/docker.sock \
+           --authorization-plugin rhel-push-plugin
+           --storage-driver devicemapper --tlsverify --tlscacert {{.AuthOptions.CaCertRemotePath}} \
+           --tlscert {{.AuthOptions.ServerCertRemotePath}} --tlskey {{.AuthOptions.ServerKeyRemotePath}} \
+           {{ range .EngineOptions.Labels }}--label {{.}} {{ end }}{{ range .EngineOptions.InsecureRegistry }}--insecure-registry {{.}} {{ end }}{{ range .EngineOptions.RegistryMirror }}--registry-mirror {{.}} {{ end }}{{ range .EngineOptions.ArbitraryFlags }}--{{.}} {{ end }}
+ExecReload=/bin/kill -s HUP $MAINPID
+MountFlags=slave
+LimitNOFILE=infinity
+LimitNPROC=infinity
+LimitCORE=infinity
+TimeoutStartSec=0
+Delegate=yes
+KillMode=process
+Environment={{range .EngineOptions.Env}}{{ printf "%q" . }} {{end}}
+
+[Install]
+WantedBy=multi-user.target
+`
+	engineConfigTemplateCentOS = `[Unit]
+Description=Docker Application Container Engine
+After=network.target rc-local.service
+
+[Service]
+Type=notify
+ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:{{.DockerPort}} -H unix:///var/run/docker.sock \
+           --storage-driver devicemapper --tlsverify --tlscacert {{.AuthOptions.CaCertRemotePath}} \
+           --tlscert {{.AuthOptions.ServerCertRemotePath}} --tlskey {{.AuthOptions.ServerKeyRemotePath}} \
+           {{ range .EngineOptions.Labels }}--label {{.}} {{ end }}{{ range .EngineOptions.InsecureRegistry }}--insecure-registry {{.}} {{ end }}{{ range .EngineOptions.RegistryMirror }}--registry-mirror {{.}} {{ end }}{{ range .EngineOptions.ArbitraryFlags }}--{{.}} {{ end }}
+ExecReload=/bin/kill -s HUP $MAINPID
+MountFlags=slave
+LimitNOFILE=infinity
+LimitNPROC=infinity
+LimitCORE=infinity
+TimeoutStartSec=0
+Delegate=yes
+KillMode=process
+Environment={{range .EngineOptions.Env}}{{ printf "%q" . }} {{end}}
+
+[Install]
+WantedBy=multi-user.target
+`
+)
+
+func init() {
+	provision.Register("minishift", &provision.RegisteredProvisioner{
+		New: func(d drivers.Driver) provision.Provisioner {
+			return NewMinishiftProvisioner("minishift", d)
+		},
+	})
+}
+
+func NewMinishiftProvisioner(osReleaseID string, d drivers.Driver) *MinishiftProvisioner {
+	systemdProvisioner := provision.NewSystemdProvisioner(osReleaseID, d)
+	systemdProvisioner.SSHCommander = provision.RedHatSSHCommander{Driver: d}
+	return &MinishiftProvisioner{
+		systemdProvisioner,
+	}
+}
+
+type MinishiftProvisioner struct {
```

Comment:
This provisioner is incomplete atm, right? AFAICT you are not implementing the complete Provisioner interface yet.